[C LIB]:Add some constant macro definitions in the libc

hongshui3000 opened the pull request:
When I was using the sof module, I found that the PI variable was missing. Hope to add some macro definitions to the minimal library。
fix https://github.com/thesofproject/sof/issues/4045
Signed-off-by: yxh <yangxiaohuamail@gmail.com>

galak reviewed (dismissed):
Can you fix the whitespace so these are aligned all the same.

andyross reviewed (commented):
Looks OK on the whole, I have no problems with including constants, even if a full libm implementation in minimal libc is clearly going to be out of scope.  This list is a little overbroad, though.  Some of these are very obscure.  Minimal libc is supposed to be minimal, and we don't want to be trying for e.g. glibc compatibility.

Maybe limit this to just the constants specified by POSIX?  c.f. https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/math.h.html

hongshui3000 commented:
> Looks OK on the whole, I have no problems with including constants, even if a full libm implementation in minimal libc is clearly going to be out of scope. This list is a little overbroad, though. Some of these are very obscure. Minimal libc is supposed to be minimal, and we don't want to be trying for e.g. glibc compatibility.
> 
> Maybe limit this to just the constants specified by POSIX? c.f. https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/math.h.html

Totally agree with this point of view

hongshui3000 committed:
libc: minimal: Add math macro definition

When the sof module code was build, it was found that
PI was not defined in the minimal library.
Here are some mathematical constant definitions to avoid build errors.

Signed-off-by: Yang XiaoHua <yangxiaohuamail@gmail.com>

hongshui3000 commented:
> Can you fix the whitespace so these are aligned all the same.

done

hongshui3000 commented:
Any comments？
Can the zephyr community accept the change?

hongshui3000 commented:
I would like to have the "stale" label removed. The issue is still important for my application.

stephanosio commented:
@cfriedt This PR is just adding some useful constants in `math.h` as defined by the POSIX standard. It would be nice to get this in for 2.7.
